wranglertools/import_data.py

```diff
@@ -1527,14 +1550,29 @@ def _verify_and_return_item(item, connection):
     return res
 
 
-def cabin_cross_check(connection, patchall, update, infile, remote, lab=None, award=None):
-    """Set of check for connection, file, dryrun, and prompt."""
+def check_and_return_input_type(inputname):
+    if pp.Path(inputname).is_file():
+        xlsx_mime = 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
+        # specific check for xlsx
+        if not mime_allowed(inputname, xlsx_mime):
+            print(f"ERROR: File {inputname} not recognized as excel file")
+            sys.exit(1)
+        return inputname, 'excel'
+    elif inputname.startswith('http'):
+        # assume a url to google sheet and look for google?
+        if 'google' not in inputname:
+            print("ERROR: URL provided does not appear to be google sheet url")
+            sys.exit(1)
+        # parse out the bookId
+        inputname = re.search("/d/([A-Za-z0-9_-]+)/*", inputname).group(1)
```

Comment:
I like constants for regex as well as they are often too complex to interpret inline, defining at top level and giving it a good name with an example or two is most helpful.